File: cpp/stubSomeValues/main/kata/BirthdayService.cpp
#include "clients/ClientRepository.h"
#include "kata/BirthdayService.h"

BirthdayService::BirthdayService(ClientRepository& withRepository) {
    repository = &withRepository;
}
  
std::string BirthdayService::greeting(std::string name) const{

    std::string greeting;

    if (repository->birthdayIsTodayFor(name)) {
        greeting = "Happy birthday " + name + "!";
    } else {
        greeting = "Good morning " + name + ".";
    }
    return greeting;
}

File: nodejs/stubSomeValues/src/BirthdayService.js
"use strict"

class BirthdayService {

    constructor(clientRepository) {
        this.repository = clientRepository
    }

    greeting(name) {

        let greeting

        const isBirthday = this.repository.birthdayIsTodayFor(name)
        if (isBirthday) {
            greeting = "Happy birthday " + name + "!"
        } else {
            greeting = "Good morning " + name + "."
        }
        return greeting

    }
}

module.exports = BirthdayService
